lighthouse-core/gather/gatherers/dom-timeline.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+// @ts-lmaonocheck - TODO: cut down on exported artifact properties not needed by audits
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+const TraceProcessor = require('../../lib/tracehouse/trace-processor.js');
+
+/* global window, MutationObserver, document, performance, getNodePath, getOuterHTMLSnippet, getNodeSelector, getNodeLabel */
+
+
+function setupObserver() {
+  // @ts-expect-error ___observedIframes does not exist on window by default
+  window.___observedIframes = [];
+  // @ts-expect-error ___observer does not exist on window by default
+  window.___observer = new MutationObserver((records) => {
+    const currTime = performance.now();
```

Comment:
did you experiment with the `performance.mark` when the mutation observer is created so we can link it up with the trace? I see the `lh_timealign` in the audit but did it get removed here?

Reply:
@patrickhulce From what I was able to gather, it seems like `performance.mark` gives the same timing. The only time it varies is when we move the call around to another line of course (though this is a small ms difference). My logic for using this is:

`performance.mark('lh_timealign');
    const markPageTs = performance.getEntriesByName('lh_timealign');
    performance.clearMarks()
    const temp = markPageTs[0].startTime;`

If we don't clear the marks, we get a big list of timings. Not sure if this would be the best way to use `performance.mark` for this case...? Wdyt?

Reply:
Agreed that `performance.mark` is not a better fit for measuring times *within* our evaluate calls (such as your example), but the benefit of `mark`/user timing is that it will show up in the trace with a trace event `ts` that is in the same timescale as all our other network request timings, FCP/LCP/TTI, etc. 

I'm not sure I currently see how `performance.now` will solve *that* problem. But maybe what you're saying is that it's already in that same timescale and I just didn't know it yet? :)

Reply:
Update for future documentation purposes: We found that userTimings and timeline events are both baselined to timeOriginEvt and converted from microseconds to ms 😁

@paulirish @patrickhulce feel free to add on/correct!